### PR TITLE
Enable none default ftp port

### PR DESCRIPTION
**Pull request by kerlion**
---
When I use ftp hook in my airflow env, it will wait till timeout when I use a none-standard ftp port. I fixed it by modify the source code. It has run for 2+ months without issue.

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by kerlion** [comment]
Is this the failed check?
Issue: [B321:blacklist] FTP-related functions are being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.

But it is ftp without TLS or ssl, we do have this scenario.

File: airflow/providers/ftp/hooks/ftp.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ def get_conn(self) -> ftplib.FTP:
         if self.conn is None:
             params = self.get_connection(self.ftp_conn_id)
             pasv = params.extra_dejson.get("passive", True)
-            self.conn = ftplib.FTP(params.host, params.login, params.password)  # nosec: B321
+            self.conn = ftplib.FTP()
```
Comment (hussein-awala) [inline airflow/providers/ftp/hooks/ftp.py]: You need to ignore the security check as we had before: `nosec: B321`

**Review by dirrao — COMMENTED** [review]
can you add the test case for the current scenario?

File: airflow/providers/ftp/hooks/ftp.py (hunk 1 above)
Comment (kerlion) [reply airflow/providers/ftp/hooks/ftp.py]: Thank you, I added it backup, I know its usage now.

Comment (kerlion) [reply airflow/providers/ftp/hooks/ftp.py]: > You need to ignore the security check as we had before: `nosec: B321`

I have fixed it.

**Review by kerlion — COMMENTED** [review]
Thanks, have fixed it.

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
This probably should have a test?

**Comment by kerlion** [comment]
Hi, is this ready for merging, anything else need I to do?

**Comment by potiuk** [comment]
> Hi, is this ready for merging, anything else need I to do?

Have you seen this @kerlion ?

> This probably should have a test?

**Review by eladkal — CHANGES_REQUESTED** [review]
Can you please add unit test to cover this change?

**Comment by kerlion** [comment]
> Can you please add unit test to cover this change?

Sorry, I do know know how to add unit test, but I tested it in my PROD env.

**Comment by eladkal** [comment]
Suppressed by https://github.com/apache/airflow/pull/39465

**Pull request closed without merging**